src/main/java/de/tum/cit/aet/artemis/programming/service/ProgrammingExerciseService.java

```diff
@@ -1087,4 +1089,38 @@ public void validateDockerFlags(ProgrammingExercise programmingExercise) {
             throw new BadRequestAlertException("The memory swap limit is invalid. The minimum memory swap limit is 0", "Exercise", "memorySwapLimitInvalid");
         }
     }
+
+    /**
+     * Find a programming exercise by its id, with eagerly loaded template and solution participation,
+     * including the latest result with feedback and test cases.
+     * <p>
+     * NOTICE: this method is quite expensive because it loads all feedback and test cases,
+     * IMPORTANT: you should generally avoid using this query except you really need all information!!
+     *
+     * @param programmingExerciseId of the programming exercise.
+     * @return The programming exercise related to the given id
+     * @throws EntityNotFoundException the programming exercise could not be found.
+     */
+    @NotNull
+    public ProgrammingExercise findByIdWithTemplateAndSolutionParticipationAndAuxiliaryReposAndLatestResultFeedbackTestCasesElseThrow(long programmingExerciseId)
+            throws EntityNotFoundException {
+        ProgrammingExercise programmingExerciseWithTemplate = programmingExerciseRepository.findWithTemplateParticipationAndSubmissionsByIdElseThrow(programmingExerciseId);
+        Set<Result> latestResultsForSubmissionsOfTemplate = resultRepository
```

Comment:
I doubt that we really need the latest result for all submissions. I am pretty sure that the latest submission with the latest result is needed

Reply:
<img width="1228" alt="image" src="https://github.com/user-attachments/assets/3b1d6469-cd51-459e-abb1-8f0cdabde863" />